components/fuse/src/main/java/org/datacleaner/components/fuse/CoalesceUnit.java

```diff
@@ -81,79 +81,88 @@ public CoalesceUnit(InputColumn<?>... inputColumns) {
      * to e.g. a copy of the input columns from a cloned job.
-     * 
+     *
      * Not doing this will result in issues such as
      * <a href="https://github.com/datacleaner/DataCleaner/issues/923">issue #923</a>
-     * 
-     * @param allInputColumns
      *
      * @return A new CoalesceUnit containing the updated columns.
      */
     public CoalesceUnit updateInputColumns(InputColumn<?>[] allInputColumns) {
-        final String[] newInputColumnNames = getInputColumnNames();
-        final InputColumn<?>[] newInputColumns = new InputColumn[newInputColumnNames.length];
-
-        for (int i = 0; i < newInputColumnNames.length; i++) {
-            boolean found = false;
-
-            final String name = newInputColumnNames[i];
+        return getUpdatedCoalesceUnit(getUpdatedInputColumns(allInputColumns, true));
+    }
 
-            // Exact match round on path.
-            for (final InputColumn<?> inputColumn : allInputColumns) {
-                if (name.contains(".") && inputColumn.isPhysicalColumn() && name
-                        .equals(inputColumn.getPhysicalColumn().getQualifiedLabel())) {
-                    newInputColumns[i] = inputColumn;
-                    found = true;
+    public InputColumn<?>[] getUpdatedInputColumns(InputColumn<?>[] allInputColumns, boolean exceptionOnMissing) {
+        final String[] newInputColumnNames = getInputColumnNames();
+        final List<InputColumn<?>> newInputColumns = new ArrayList<>(newInputColumnNames.length);
+
+        for (final String newInputColumnName : newInputColumnNames) {
+            final InputColumn<?> updatedInputColumn = updateInputColumn(allInputColumns, newInputColumnName);
+            if (updatedInputColumn == null) {
+                if (exceptionOnMissing) {
+                    final List<String> names =
+                            Arrays.stream(allInputColumns).map(InputColumn::getName).collect(Collectors.toList());
+                    throw new CoalesceUnitMissingColumnException(this, newInputColumnName,
+                            "Column '" + newInputColumnName + "' not found. Available columns: " + names);
                 }
+            } else {
+                newInputColumns.add(updatedInputColumn);
             }
+        }
+        return newInputColumns.toArray(new InputColumn[newInputColumns.size()]);
+    }
 
-            if (!found) {
-                // Trimmed and case-insensitive path match round.
-                for (final InputColumn<?> inputColumn : allInputColumns) {
-                    if (name.contains(".") && inputColumn.isPhysicalColumn() && name.trim()
-                            .equalsIgnoreCase(inputColumn.getPhysicalColumn().getQualifiedLabel())) {
-                        newInputColumns[i] = inputColumn;
-                        found = true;
-                    }
-                }
+    private InputColumn<?> updateInputColumn(final InputColumn<?>[] allInputColumns, final String newInputColumnName) {
```

Comment:
The name of this method is very confusing to me. The name seems to indicate that an InputColumn is updated, but this is not the case, an InputColumn is returned based on its name and its availability in the passed InputColumn array. It seems more logical to me if the method is called "getInputColumn" and the second argument omits the "new", because for the scope of this method it is not relevant whether or not it is the name of a new InputColumn.